### PR TITLE
provider/datadog: Improve datadog timeboard support

### PR DESCRIPTION
**Pull request by nyanshak**
* Add more complete support for Datadog timeboards, now including (but not limited to) conditional formatting, events, and more.

### CODE REVIEW

**Comment by nyanshak** [comment]
Right now this is based off my fork of go-datadog-api, because I'm waiting on some changes to make it through approvals on that end. Plan is to redo first commit based on update to go-datadog-api when it's merged upstream.

I have tested on a dashboard including tons of strange formatting without issue.

Looking for any feedback on changes in advance of upstream changes to the go-datadog-api lib.

**Comment by nyanshak** [comment]
Change merged into go-datadog-api library, and now I reworked this PR based on that.

**Comment by wolfspyre** [comment]
ooh! this would be awesome! +1 
What needs to happen to get this merged?

File: builtin/providers/datadog/resource_datadog_monitor.go [hunk 1]
```diff
@@ -393,16 +393,9 @@ func resourceDatadogMonitorDelete(d *schema.ResourceData, meta interface{}) erro
 	return nil
 }
 
-func resourceDatadogImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```
Comment (ojongerius) [inline builtin/providers/datadog/resource_datadog_monitor.go]: Could you give some context as to why you removed this? `ImportStatePassthrough` just does a return of `[]*schema.ResourceData{d}` and `nil`. What happens when the monitor was manually removed a user?

Comment (nyanshak) [inline builtin/providers/datadog/resource_datadog_monitor.go]: I think I didn't quite understand importing and this should be changed back. I'm still somewhat new to terraform and the importing functionality was a bit confusing to me.

**Comment by nyanshak** [comment]
I added a new test. @ojongerius / @stack72 would you mind reviewing?

**Comment by ojongerius** [comment]
@nyanshak awesome! LGTM.

File: builtin/providers/datadog/resource_datadog_timeboard_test.go [hunk 2]
```diff
@@ -62,6 +62,56 @@ resource "datadog_timeboard" "acceptance_test" {
 }
 `
 
+const config3 = `
+resource "datadog_timeboard" "acceptance_test" {
+  title = "Acceptance Test Timeboard"
+  description = "Created using the Datadog prodivider in Terraform"
```
Comment (nyanshak) [inline builtin/providers/datadog/resource_datadog_timeboard_test.go]: That's what I get for copy-pasting from the existing tests 👎 but I fixed it now.

**Comment by nyanshak** [comment]
@stack72 / @jen20 thoughts? 👍 / 👎

**Comment by stack72** [comment]
This LGTM! Sorry it took so long to get back to it :)

```
% make testacc TEST=./builtin/providers/datadog                                                                                                                                                       ✹
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/12/05 13:07:24 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/datadog -v  -timeout 120m
=== RUN   TestDatadogMonitor_import
--- PASS: TestDatadogMonitor_import (81.40s)
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccDatadogMonitor_Basic
--- PASS: TestAccDatadogMonitor_Basic (79.46s)
=== RUN   TestAccDatadogMonitor_Updated
--- PASS: TestAccDatadogMonitor_Updated (79.75s)
=== RUN   TestAccDatadogMonitor_TrimWhitespace
--- PASS: TestAccDatadogMonitor_TrimWhitespace (84.41s)
=== RUN   TestAccDatadogMonitor_Basic_float_int
--- PASS: TestAccDatadogMonitor_Basic_float_int (106.20s)
=== RUN   TestAccDatadogTimeboard_update
--- PASS: TestAccDatadogTimeboard_update (68.86s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/datadog	500.100s
```

**Pull request merged**